pydoctor/templatewriter/writer.py

```diff
@@ -56,13 +57,28 @@ def prepOutputDirectory(self) -> None:
         """
         Write static CSS and JS files to build directory.
         """
-        os.makedirs(self.base, exist_ok=True)
-        for template in self.template_lookup.templates:
-            if isinstance(template, _StaticTemplate):
-                with self.base.joinpath(template.name).open('w', encoding='utf-8') as jobj:
-                    jobj.write(template.text)
+        os.makedirs(self.output_dir, exist_ok=True)
+        self._writeStaticTemplates(self.template_lookup.templates)
+    
+    def _writeStaticTemplates(self, templates: Iterable[Template], subfolder: Optional[str] = None) -> None:
+        """
+        Write all L{_StaticTemplate} to output directory, inspect L{_TemplateSubFolder} 
+        and reccursively write the static templates in subfolders.
+        """
+        _template_rel_path_t = f"{subfolder + os.sep if subfolder else ''}%s"
```

Comment:
This is risky: if the path contains percent characters, the interpolation could do the wrong thing.

Instead, I think you could take `self.output_dir` if `subfolder` is empty and look up the subdir otherwise. So operate on `Path` objects instead of on strings.